src/main/java/CheonsaBot.java

```diff
@@ -0,0 +1,421 @@
+                break;
+            case "bye":
+                getBye();
+                return false;
+            case "list":
+                printTaskList();
+                break;
+            case "delete":
+                deleteTask(argument);
+                break;
+            default:
+                addTask(command, words);
+                break;
+        }
+        return true;
+    }
+
+    /**
+     * Marks a task as completed based on the provided task number.
+     *
+     * @param taskNumber The number of the task to be marked as completed.
+     */
+    private static void markTask(String taskNumber) {
+        try {
+            int index = Integer.parseInt(taskNumber) - 1;
+            if (index >= 0 && index < tasks.size()) {
+                tasks.get(index).setAsDone();
+                updateTaskFile();
+                System.out.println(getHorizontalLine());
+                System.out.println("Marked task as done: " + tasks.get(index));
+                System.out.println(getHorizontalLine());
+            } else {
+                System.out.println(getHorizontalLine());
+                System.out.println("Task number out of range :(");
+                System.out.println("Current number of tasks: " + tasks.size());
+                System.out.println(getHorizontalLine());
+            }
+            
+        } catch (NumberFormatException e) {
+            System.out.println(getHorizontalLine());
+            System.out.println("Invalid task number :(");
+            System.out.println(getHorizontalLine());
+        }
+    }
+
+    /**
+     * Unmarks a task as incomplete based on the provided task number.
+     *
+     * @param taskNumber The number of the task to be unmarked.
+     */
+    private static void unmarkTask(String taskNumber) {
+        try {
+            int index = Integer.parseInt(taskNumber) - 1;
+            if (index >= 0 && index < tasks.size()) {
+                tasks.get(index).setAsUndone();
+                updateTaskFile();
+                System.out.println(getHorizontalLine());
+                System.out.println("Unmarked task: " + tasks.get(index));
+                System.out.println(getHorizontalLine());
+            } else {
+                System.out.println(getHorizontalLine());
+                System.out.println("Task number is out of range :(");
+                System.out.println(getHorizontalLine());
+            }
+        } catch (NumberFormatException e) {
+            System.out.println(getHorizontalLine());
+            System.out.println("Invalid task number :(");
+            System.out.println(getHorizontalLine());
+        }
+    }
+
+    /**
+     * Adds a new task based on the command and argument provided.
+     *
+     * @param command The type of task to add (e.g., "todo", "deadline", "event").
+     * @param words The details of the task.
+     */
+    private static void addTask(String command, String[] words) {
```

Comment:
This method is pretty long, you may consider extracting each case into a separate method. It will be better to keep methods short.